### DISCUSSION TITLE
Import data with timestamp?

### DISCUSSION THREAD

**Discussion by adamwolf** [plausible/analytics#4129] [Self-Hosted Support] [answered]
Hi folks!

I am trying to import data.  Right now, I'm using the Events API.  Does the Events API assume the event is happening "right now"? Is there a way to give it a different timestamp?  Is there a better way to feed Plausible non-live data?

Thanks!

**Comment by ruslandoga** [comment, marked as answer]
👋 @adamwolf 

> Does the Events API assume the event is happening "right now"?

Yes: https://github.com/plausible/analytics/blob/c81cb1693320b35a0d4a1fddbcb5d7bb7cbcf9bf/lib/plausible/ingestion/request.ex#L60

> Is there a way to give it a different timestamp?

Via an HTTP request to /api/event, no. 

> Is there a better way to feed Plausible non-live data?

Depending on your use-case CSV imports might be useful: https://plausible.io/docs/csv-import - available starting in [v2.1.0](https://github.com/plausible/analytics/discussions/4125)

**Reply by adamwolf** [reply]
Thanks for this!

I rigged up a prototype.  I am trying to ingest some request logs from an environment where many folks aren't using Javascript.  I'm also, sigh, on Cloudflare, where getting logs requires an enterprise account. I'm currently putting request information into their sqlite-alike using a Worker and then periodically exporting from the database to Plausible.

That CSV import must be new! I don't think it existed last week when I put this all together :) Not having filters is a bit of a bummer, so I think I'll rekajigger the prototype to push pageviews to Plausible "live".
